core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java

```diff
@@ -50,19 +52,20 @@
  *
  * Created by gupele on 12/18/2014.
  */
-public final class TransmissionNetworkOutput implements TransmissionOutput {
-    private final static String CONTENT_TYPE_HEADER = "Content-Type";
-    private final static String CONTENT_ENCODING_HEADER = "Content-Encoding";
-    private final static String RESPONSE_THROTTLING_HEADER = "Retry-After";
+public final class TransmissionNetworkOutput implements ConfiguredTransmissionOutput {
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String CONTENT_ENCODING_HEADER = "Content-Encoding";
+    private static final String RESPONSE_THROTTLING_HEADER = "Retry-After";
 
-    public final static String DEFAULT_SERVER_URI = "https://dc.services.visualstudio.com/v2/track";
+    public static final String DEFAULT_SERVER_URI = "https://dc.services.visualstudio.com/v2/track";
 
     // For future use: re-send a failed transmission back to the dispatcher
     private TransmissionDispatcher transmissionDispatcher;
 
-    private final String serverUri;
+    private String serverUri;
 
     private volatile boolean stopped;
+    private final AtomicReference<TelemetryConfiguration> configuration = new AtomicReference<>();
```

Comment:
i think use `volatile` instead of `AtomicReference`, since not doing any atomic operations on it